### DISCUSSION TITLE
Running BIS3.0 Schema/Business Validations before Sending to AP

### DISCUSSION THREAD

**Discussion by anudhij** [OxalisCommunity/Oxalis-AS4#224] [Q&A]
We are in the process of running all the tests from OpenPeppol Test Bed. We have successfully completed the tests for Standalone AP. While running the tests for **BIS testing** we are able to successfully complete the tests with Valid Payload, but the tests with Invalid Payload having validation errors are getting failed.

Peppol Testbed shows the below reason for test failure:

> Received transaction with instance identifier 'POP000602-32-20231102T123703' was not expected to be received because it was generated with validation errors. Your AP implementation MUST be able to identify these situations and never initiate transactions towards other APs with invalid files. Restart the test case to continue.

I am using Docker command to send the message to TestBed AP

> docker run -e "MODE=standalone" 70f35f3fe3c4 -f TestFile_001__BISv3_Invoice.xml

Am I missing some parameter required to be passed for considering the validations before sending?

**Comment by dladlk** [comment]
Hello @anudhij , Oxalis does not provide schema/schematron validation services, it is a job for its users to implement required validations before sending. There are multiple reasons for this - starting from keeping validation rules actual. So to pass this part of test, you just had not to send anything back to TestBed :)

**Comment by anudhij** [comment]
Sure. Thanks @dladlk !!
